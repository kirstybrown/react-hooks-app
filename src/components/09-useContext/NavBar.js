import React from 'react';
import { Link, NavLink } from 'react-router-dom';

const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">

      <Link to="/" className="navbar-brand" > useContext </Link>
    
      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        <div className="navbar-nav">
         <NavLink activeClassName="active" to="/" className="nav-link" aria-current="page" > Home </NavLink>
         <NavLink activeClassName="active" to="/about" className="nav-link" > About </NavLink>
         <NavLink activeClassName="active" to="/login" className="nav-link" > Login </NavLink>
        </div>
      </div>
  </nav>
  )
}

export default NavBar