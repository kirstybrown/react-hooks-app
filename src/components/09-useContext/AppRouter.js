import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route
  } from "react-router-dom";
  import AboutScreen from './AboutScreen';
  import LoginScreen from './LoginScreen';
  import HomeScreen from './HomeScreen';
import NavBar from './NavBar';
  

const AppRouter = () => {
  return (
      <Router>
          <div>

              <NavBar />
                <div className='container'>
                    <Routes>
                        <Route path="/" element={ <HomeScreen/> } />
                        <Route path="/about" element={ <AboutScreen/> } />
                        <Route path="/login" element={ <LoginScreen/> } />

                        <Route element={ <HomeScreen /> } />
                    </Routes>
                </div>
              
          </div>
      </Router>
    
  )
}

export default AppRouter