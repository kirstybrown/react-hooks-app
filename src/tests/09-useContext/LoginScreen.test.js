import { mount } from 'enzyme';

import LoginScreen from "../../components/09-useContext/LoginScreen";
import { UserContext } from '../../components/09-useContext/UserContext';


describe('Tests for LoginScreen', () => { 


    const setUser = jest.fn();

    const wrapper = mount( <UserContext.Provider value={{
        setUser
    }}>
        <LoginScreen /> 
    </UserContext.Provider>
    );

    test('should show component correctly', () => { 
        expect( wrapper ).toMatchSnapshot();
     });

    test('should execute setUser with the expected argument', () => { 

        wrapper.find('button').prop('onClick')();

        expect( setUser ).toHaveBeenCalledWith({
            id: 123,
            name: 'Kirsty'
          });
      });
});