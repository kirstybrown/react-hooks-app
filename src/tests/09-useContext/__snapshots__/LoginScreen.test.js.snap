// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for LoginScreen should show component correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
